### PR TITLE
JBEAP-6177 : Reconnect handlers leaking channels, need to unregister the ejb receiver

### PR DESCRIPTION
**Pull request by panossot**
[JBEAP-6177] : https://issues.jboss.org/browse/JBEAP-6177

### CODE REVIEW

File: src/main/java/org/jboss/ejb/client/EJBClientContext.java [hunk 1]
```diff
@@ -153,6 +153,18 @@ private void init(ClassLoader classLoader) {
     }
 
     /**
+     * Returns the Ejb Receivers of the current EjbClientContext.
+     */
+    public EJBReceiver[] getEjbReceivers() {
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: This API change is not allowed.

File: src/main/java/org/jboss/ejb/client/EJBClientContext.java [hunk 2]
```diff
@@ -475,6 +487,20 @@ public void run() {
         }
     }
 
+    public void unregisterNodeEJBReceivers(String nodeName) {
```
Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: This is another API change that we can't include.  Essentially we cannot be adding public methods at this point, without a very good justification.

**Comment by dmlloyd** [comment]
Why does this PR reference JBEAP-6177 whereas #183 references JBEAP-6554 with the same title?

File: src/main/java/org/jboss/ejb/client/EJBClientContext.java (hunk 2 above)
Comment (panossot) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: It is similar to this though : https://github.com/rstancel/jboss-ejb-client/blob/477e82d5e0c4adcfe42f507465c0531e9623c245/src/main/java/org/jboss/ejb/client/ClusterContext.java#L414

Comment (panossot) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: Or better, similar to this : https://github.com/jbossas/jboss-ejb-client/blob/2.1/src/main/java/org/jboss/ejb/client/ClusterContext.java#L414

Comment (dmlloyd) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: Did that get released in anything?

Comment (panossot) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: Could be in anything that uses 1.0.38.Final.

Comment (panossot) [inline src/main/java/org/jboss/ejb/client/EJBClientContext.java]: ** 2.1.7.Final

**Comment by panossot** [comment]
Closing this PR. 
Replaced via https://github.com/jbossas/jboss-ejb-client/pull/201

**Pull request closed without merging**